File: backend/src/utils/platformSlugMap.js
// BACKEND: src/utils/platformSlugMap.js

function normalizeKey(s) {
    return String(s || "")
        .toLowerCase()
        .trim()
        .replace(/\s+/g, " ")
        .replace(/[^a-z0-9 ]/g, "");
}

const ALIASES = {
    netflix: "netflix",
    spotify: "spotify",
    chatgpt: "chatgpt",
    prime: "prime",
};

function toCodeSlug(platformSlugFromDBOrRequest) {
    const key = normalizeKey(platformSlugFromDBOrRequest);

    // ✅ CHATGPT
    if (key.includes("chatgpt") || key.includes("chat gpt")) return "chatgpt";

    // ✅ SPOTIFY (spotify-3m, spotify premium, etc)
    if (key.includes("spotify")) return "spotify";

    // ✅ PRIME VIDEO (prime video, amazon prime, primevideo, etc)
    if (key.includes("prime")) return "prime";

    // ✅ NETFLIX
    if (key.includes("netflix")) return "netflix";

    return ALIASES[key] || key.replace(/\s/g, "");
}

module.exports = { toCodeSlug };

File: go-backend/codes-service/utils/platformSlugMap.go
package utils

import (
	"regexp"
	"strings"
)

var normalizeRe1 = regexp.MustCompile(`\s+`)
var normalizeRe2 = regexp.MustCompile(`[^a-z0-9 ]`)

func normalizeKey(s string) string {
	lower := strings.ToLower(strings.TrimSpace(s))
	spaced := normalizeRe1.ReplaceAllString(lower, " ")
	return normalizeRe2.ReplaceAllString(spaced, "")
}

func ToCodeSlug(platformSlug string) string {
	key := normalizeKey(platformSlug)

	if strings.Contains(key, "chatgpt") || strings.Contains(key, "chat gpt") {
		return "chatgpt"
	}
	if strings.Contains(key, "spotify") {
		return "spotify"
	}
	if strings.Contains(key, "prime") {
		return "prime"
	}
	if strings.Contains(key, "netflix") {
		return "netflix"
	}

	aliases := map[string]string{
		"netflix": "netflix",
		"spotify": "spotify",
		"chatgpt": "chatgpt",
		"prime":   "prime",
	}

	if alias, exists := aliases[key]; exists {
		return alias
	}

	return strings.ReplaceAll(key, " ", "")
}

func NormalizeSlug(s string) string {
	return strings.ToLower(strings.TrimSpace(s))
}
